### DISCUSSION TITLE
Can MQ exception alarms be defined?

### DISCUSSION THREAD

**Discussion by Mxgxg** [apache/skywalking#8213] [Q&A] [answered]
Consume MQ messages asynchronously, tracing the process through a manual API.

It doesn't belong to any endpoint. Is there a way to trigger an alarm if something goes wrong?

![image](https://user-images.githubusercontent.com/79822553/144004129-51b0f059-e8aa-4b74-a1c3-e29ddde75a37.png)

**Comment by Mxgxg** [comment, marked as answer]
If it is not displayed in the endpoint search bar, it is not considered an endpoint, but only a span?

**Reply by wu-sheng** [reply]
Basic concepts, only the entry span represents an endpoint.

**Reply by Mxgxg** [reply]
I read the document, mq consumers will think it is EntrySpan, which is an endpoint, but LocalSpan is displayed here. Not very understanding

**Reply by wu-sheng** [reply]
It is entry span. If you are looking at local span, which means you are looking for a wrong span.

**Reply by Mxgxg** [reply]
Okay, is there any way for this exception to trigger an alarm? I did not find a solution after reading the document. Can you provide some related articles, thank you very much

**Reply by wu-sheng** [reply]
Alarm is for metrics. No one sets an alarm rule for tracing, this will kill your notification system or your mailbox for sure.
You should consider more before ask for something.

**Reply by Mxgxg** [reply]
Sorry, I didn't express it clearly. I set metrics for endpoint exceptions, and set alarm thresholds in the alarm configuration, and the test is effective. 
But the span here does not represent the endpoint, and the UI does not have any indicators to detect this anomaly, so I don't know which metric should be used to count it.

![image](https://user-images.githubusercontent.com/79822553/144017806-8747bce2-7e4e-41c4-b370-016e4cadcc73.png)
![image](https://user-images.githubusercontent.com/79822553/144017988-5b8d3657-da64-4a69-8226-b875e7d8c81a.png)

**Reply by wu-sheng** [reply]
You could set up logic endpoint flag to the span, then it would be treated as an custom endpoint.

**Reply by Mxgxg** [reply]
Where is the logical endpoint configured? The location of the configuration file was not found. ^ ^

**Reply by wu-sheng** [reply]
https://skywalking.apache.org/docs/skywalking-java/v8.8.0/en/setup/service-agent/java-agent/logic-endpoint/

Please read document before ask. You are saving your time by costing mine. With the latency you asked, you even don't try to find.
This is a simple search on the website.

**Reply by Mxgxg** [reply]
Brother Sheng, I'm sorry to delay your time, I think you have misunderstood. The document has been found long ago, and I don't understand where to configure it. 
Is it to use @Tag annotation?

**Reply by wu-sheng** [reply]
Anyway you like, tag is tag. Nothing special.

**Reply by Mxgxg** [reply]
solved. Thank you very much for your patient answers. Caused you trouble
